File: realisations/PHP/validate.php
#!/usr/bin/env php

<?php

require('vendor/autoload.php');

$input = json_decode($argv[1], true);
$rules = json_decode($argv[2], true);

$validator = new \Validator\LIVR( $rules );

$output    = $validator->validate($input, $rules);
$errors    = $validator->getErrors();

print json_encode($errors
    ? [ 'errors' => $errors ]
    : [ 'output' => $output ]
);

?>

File: implementations/Erlang/validate.erl
#!/usr/bin/env escript
%%! -pa deps/olifer/ebin deps/parselib/ebin deps/jsx/ebin -sname livr_playground

main([InputJson, RulesJson])->
    ok = olifer:start(),
    try
        Input = jsx:decode((list_to_binary(InputJson))),
        Rules = jsx:decode((list_to_binary(RulesJson))),

        JsonResult = case olifer:validate(Input, Rules) of
            {ok, Result} -> binary_to_list(jsx:encode([{<<"output">>, Result}]));
            {errors, Result} -> binary_to_list(jsx:encode([{<<"errors">>, Result}]))
        end,
        io:fwrite(JsonResult)
    catch
    _:_ ->
        io:format(standard_error, "~p~n", [erlang:get_stacktrace()]),
        exit("Olifer internal error")
    end.